tests/unit/test_collector.py

```diff
@@ -41,3 +45,182 @@ def test_01_mega_raid_collector_installed_and_okay(self):
+        lsi_sas_2_collector = LSISAS2ControllerCollector()
+        lsi_sas_2_collector.sas2ircu = Mock()
+        mock_adapters = {
+            "0": {
+                "Index": "0",
+                "Adapter Type": "SAS2008",
+                "Vendor ID": "1000h",
+                "Device ID": "72h",
+                "Pci Address": "00h:05h:00h:00h",
+                "SubSys Ven ID": "1028h",
+                "SubSys Dev ID": "1f1eh",
+            },
+        }
+        mock_information = {
+            "controller": {
+                "Controller type": "SAS2008",
+                "BIOS version": "7.11.10.00",
+                "Firmware version": "7.15.08.00",
+                "Channel description": "1 Serial Attached SCSI",
+                "Initiator ID": "0",
+                "Maximum physical devices": "39",
+                "Concurrent commands supported": "2607",
+                "Slot": "0",
+                "Segment": "0",
+                "Bus": "5",
+                "Device": "0",
+                "Function": "0",
+                "RAID Support": "Yes",
+            },
+            "ir_volumes": {
+                "1": {
+                    "Volume ID": "286",
+                    "PI Supported": "Yes",
+                    "PI Enabled": "Yes",
+                    "Status of volume": "Okay (OKY)",
+                    "Volume wwid": "0677c0fb06777e7b",
+                    "RAID level": "RAID1",
+                    "Size (in MB)": "139236",
+                    "Boot": "Primary",
+                    "Physical hard disks": {
+                        "PHY[0] Enclosure#/Slot#": "1:0",
+                    },
+                }
+            },
+            "enclosures": {
+                "1": {
+                    "Enclosure#": "1",
+                    "Logical ID": "5782bcb0:19e35100",
+                    "Numslots": "9",
+                    "StartSlot": "0",
+                }
+            },
+            "physical_disks": {
+                "1:0": {
+                    "Enclosure #": "1",
+                    "Slot #": "0",
+                    "SAS Address": "4433221-1-0700-0000",
+                    "State": "Ready (RDY)",
+                    "Size (in MB)/(in sectors)": "476940/976773167",
+                    "Manufacturer": "ATA",
+                    "Model Number": "ST9500620NS",
+                    "Firmware Revision": "AA0D",
+                    "Serial No": "9XF3RK3W",
+                    "GUID": "5000c5007bc1cfd7",
+                    "Protocol": "SATA",
+                    "Drive Type": "SATA_HDD",
+                },
+            },
+        }
+
+        lsi_sas_2_collector.sas2ircu.get_adapters.return_value = mock_adapters
+        lsi_sas_2_collector.sas2ircu.get_all_information.return_value = mock_information
+
+        payloads = lsi_sas_2_collector.collect()
+
+        available_metrics = [spec.name for spec in lsi_sas_2_collector.specifications]
```

Comment:
I'm thinking is it reasonable to fetch the specification from the source code. 
In the most primitive way, we should declare the expected specification here and verify if the output equals to the specification.

But I'm okay with it now, but we need to review the the specification in the source code to confirm it is declared as the detail design specification.